### ISSUE TITLE
Apdex can't read processor properly

### ISSUE THREAD

**Issue by lfox91** [artilleryio/artillery#2193]
Version info: 
```
Artillery: 2.0.0-37
Node.js:   v18.17.1
OS:        darwin

-> npm -g ls
├── artillery-plugin-apdex@1.0.0
├── artillery-plugin-ensure@1.2.0
├── artillery-plugin-metrics-by-endpoint@1.2.0
├── artillery@2.0.0-37
├── corepack@0.15.1
├── npm@9.6.7
├── ts-node@10.9.1
└── typescript@5.2.2

```

Running this command: 
```
artillery run activate-login/activate-login.yml
```

I expected to see this happen:

The test run

Instead, this happened:

```
TypeError: Cannot create property 'apdexAfterResponse' on string './activate-login.js'

=======
class ApdexPlugin {
  constructor(script, _events) {
    this.script = script;

    const t = script.config.apdex?.threshold || script.config.plugins.apdex?.threshold || 500;
    console.log("!!!!\n",script.config.processor,"\n!!!")  // Prints: ./activate-login.js
    if (!script.config.processor) {
      script.config.processor = {};
    }

    script.scenarios.forEach(function (scenario) {
      scenario.afterResponse = [].concat(scenario.afterResponse || []);
      scenario.afterResponse.push('apdexAfterResponse');
    });

    function apdexAfterResponse(req, res, userContext, events, done) {
      const total = res.timings.phases.total;
      if (total <= t) {
        events.emit('counter', METRICS.satisfied, 1);
      } else if (total <= 4 * t) {
        events.emit('counter', METRICS.tolerated, 1);
      } else {
        events.emit('counter', METRICS.frustrated, 1);
      }

      return done();
    }

    // processor is being interpreted as a string which can't have a property
    script.config.processor.apdexAfterResponse = apdexAfterResponse;
                              ^^^^^^^^

```
The ApdexPlugin seems unable to treat the processor as an object as it's expecting. Everything works if I comment out Apdex in the config.

Files being used:
```
//activate-login.yml
config:
  target: [redacted]
  processor: ./activate-login.js
  tls:
    rejectUnauthorized: true
  http:
    timeout: 10
    maxSockets: 5
  phases:
    - duration: 5
      arrivalRate: 1
      rampTo: 2
      name: Warm up phase
    # - duration: 60
    #   arrivalRate: 5
    #   rampTo: 10
    #   name: Ramp up load
    # - duration: 30
    #   arrivalRate: 10
    #   rampTo: 30
    #   name: Spike phase
  plugins:
    ensure: {}
    apdex: {}
    metrics-by-endpoint: {}
  apdex:
    threshold: 100
  ensure:
    thresholds:
      - http.response_time.p99: 100
      - http.response_time.p95: 75
scenarios:
  - flow:
      - log: "Current env: {{$environment}}"
      - function: "loginFunc"
      - log: "[redacted]"
      - get:
          URL: "[redacted]={{[redacted]}}" 
          capture: { json: "$.token", as: "authToken" }
      - log: "Retrieved token from /sims: {{authToken}}"
      
// activate-login.js
async function loginFunc(context, event, done) {
  // iccid payload
  const body = JSON.stringify({
     [redacted]
  });
  const method = "POST";
  const headers = {
    Accept: "application/json",
    "Content-Type": "application/json",
  };

  const url = "[redacted]";
  try {
    const response = await fetch(url, { body, method, headers });
    const jsonData = await response.json();
    console.log(jsonData);
    context.var = { ...context.var, actCode: jsonData.prop };
  } catch (error) {
    console.error(error);
    throw error;
  } finally {
    done();
  }
}

module.exports = { loginFunc };
```

Other notes:
I tried installing the plugins directly to fix the issue - this seems like a legacy way of handling plugins. It's unclear to me when reading through the docs, please let me know if this is still necessary. Really enjoying the ease of setup otherwise.

**Comment by bernardobridge** [comment]
Hey @lfox91 👋  ! Thanks for taking the time to dig through this, and for the thorough bug report 🙇‍♂️ - very helpful!

I can confirm this is a bug on the Apdex plugin logic. I'll work on a fix for this tomorrow, as I believe I know the issue. Will let you know when we have a canary version with a fix. Thanks!

**Comment by bernardobridge** [comment]
Hi @lfox91.

The canary version `2.0.0-37-9c51118` should include the fix to this issue, so if you need this, you can try using this version for now. We'll try to do a version release hopefully later this week and it will land also on Artillery v38.

Thanks again!

**Comment by lfox91** [comment]
It worked great! Thank you all!

**Comment by bernardobridge** [comment]
A fix for this is now released on [Artillery v2.0.0-38](https://github.com/artilleryio/artillery/releases/tag/v2.0.0-38).
